addons/l10n_ro_edi_stock/models/l10n_ro_edi_stock_etransport_interface.py

```diff
@@ -0,0 +1,640 @@
+                elif len(missing_field_names) > 1:
+                    errors.append(_("%(location_group)s is missing following fields: %(field_names)s", location_group=loc_group, field_names=missing_field_names))
+
+        # carrier partner fields
+        if self.l10n_ro_edi_stock_carrier_id.l10n_ro_edi_stock_partner_id.country_id.code != 'RO':
+            errors.append(_("The delivery carrier partner has to be located in Romania."))
+        else:
+            missing_carrier_partner_fields = []
+            partner = self.l10n_ro_edi_stock_carrier_id.l10n_ro_edi_stock_partner_id
+
+            if not partner.city:
+                missing_carrier_partner_fields.append(_("City"))
+
+            if not partner.street:
+                missing_carrier_partner_fields.append(_("Street"))
+
+            if len(missing_carrier_partner_fields) == 1:
+                errors.append(_("The delivery carrier partner is missing the %(field_name)s field.", field_name=missing_carrier_partner_fields[0]))
+            elif len(missing_carrier_partner_fields) > 1:
+                errors.append(_("The delivery carrier partner is missing following fields: %(field_names)s", field_names=', '.join(missing_carrier_partner_fields)))
+
+        if not self.l10n_ro_edi_stock_company_id.l10n_ro_edi_access_token:
+            errors.append(_('Romanian access token not found. Please generate or fill it in the settings.'))
+
+        return errors
+
+    def _check_fetch_values(self, errors=None) -> list:
+        if errors is None:
+            errors = []
+        self.ensure_one()
+
+        if not self.l10n_ro_edi_stock_company_id.l10n_ro_edi_access_token:
+            errors.append(_('Romanian access token not found. Please generate or fill it in the settings.'))
+            return errors
+
+        match self.l10n_ro_edi_stock_state:
+            case 'sending_failed':
+                if self._get_last_sent_document():
+                    errors.append(_("This document has not been successfully sent yet because it contains errors."))
+                else:
+                    errors.append(_("This document has not been corrected yet because it contains errors."))
+            case 'sent':
+                errors.append(_("This document has already been successfully sent to anaf."))
+
+        return errors
+
+    # Template Helpers
+
+    def _get_declarant_ref(self) -> str:
+        self.ensure_one()
+        return self.picking_id.name
+
+    def _get_scheduled_date(self):
+        self.ensure_one()
+        return self.picking_id.scheduled_date.date()
+
+    def _get_commercial_partner(self):
+        self.ensure_one()
+        return self.picking_id.partner_id.commercial_partner_id
+
+    def _get_transport_partner(self):
+        self.ensure_one()
+        return self.picking_id.carrier_id.l10n_ro_edi_stock_partner_id
+
+    # Document Helpers
```

Comment:
To make it clear (and cool), let's add some borders to these class section comments :)
```suggestion
    ################################################################################
    # Document Helpers
    ################################################################################
```

(do the same for all other)